decode.go

```diff
@@ -58,7 +58,7 @@ func Decode(dst, src []byte) ([]byte, error) {
 	}
 
 	if !isMLZ {
-		if l, _ := s2.DecodedLen(block); l > MaxBlockSize {
```

Comment:
Why not let `s2.Decode()` handle it?

Reply:
Mainly because I don't want to forward another error type. So I will have to handle it separately anyway.